compressor/management/commands/compress.py

```diff
@@ -284,5 +291,45 @@ def handle(self, **options):
         self.compress(sys.stdout, **options)
 
 
+def _do_render_task(index):
+    global tasks
+    task = tasks[index]
+    try:
+        return _do_render(*task)
+    except Exception as e:
+        log = task[0]._log
+        log.write(traceback.format_exc())
+        raise
+
+
+def _do_render(template, parser, node, context_dict, manifest, manifest_lock):
+    init_context = parser.get_init_context(context_dict)
+    context = Context(dict_=init_context)
+
+    rendered = parser.render_nodelist(template, context, node)
+    key = get_offline_hexdigest(rendered)
+
+    if key in manifest:
+        return None
```

Comment:
do you know whether this is a vital optimization? i.e., if it doesn't run much slower without this, we could just remove this and the lock altogether, let the tasks return tuples like `(key, result)` and build the dictionary in serial afterwards